Sourcery/SourceryTemplate.swift

```diff
@@ -44,3 +66,44 @@ private func stringContentFilter(_ name: String, _ filter: @escaping (String) ->
         return filter(s)(arg)
     }
 }
+
+private struct Filter<T> {
+    static func make(_ filter: @escaping (T) -> Bool) -> (Any?) throws -> Any? {
+        return { (any) throws -> Any? in
+            switch any {
+            case let type as T:
+                return filter(type)
+
+            case let array as NSArray:
+                return array.flatMap { $0 as? T }.filter(filter)
+
+            default:
+                return any
+            }
+        }
+    }
+}
+
+private struct FilterOr<T, Y> {
```

Comment:
I wonder if using `Either<A, B>` would work here and would be more scalable, for example later if we add `isClass` property for variables. But we can try that later =)